app/controllers/application_controller.rb

```diff
@@ -1,3 +1,13 @@
 class ApplicationController < ActionController::Base
+  include Pundit
   protect_from_forgery with: :exception
+
+  rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
+
+  private
+
+  def user_not_authorized
+    flash[:danger] = 'You are not authorized to perform this action.'
+    redirect_to(request.referer || root_path)
```

Comment:
You can refactor it to:
```ruby
redirect_back(fallback_location: root_path)
```
It does exactly the same, but I believe that it's a bit more readable ;)